### ISSUE TITLE
Пропала возможность подключить аккаунт Google для синхронизации

### ISSUE THREAD

**Issue by UnclWish** [deemru/Chromium-Gost#70]
После обновления до 114.0.5735.110 в браузере пропала возможность подключить аккаунт google для синхронизации.
На значке пользователя рядом с меню браузера серая иконка с подсказкой "Вы", по нажатию на которую появляется меню пользователя "Пользователь 1". Кнопки включения синхронизации с входом в google нет.
Нет кнопки включения синхронизации и в настройках, в разделе "Я и Google".

После отката до версии 114.0.5735.90 всё вернулось обратно. Подключил свой аккаунт и включилась синхронизация.

Снова попробовал обновиться до 110-й версии. После обновления история повторилась - синхронизация пропала, кнопок включения и входа в свой google-аккаунт нигде нет.

**Comment by deemru** [comment]
Видимо вы продвинутый пользователь, расскажите, как вам удавалось раньше подключать синхронизацию?

Через вот этот трюк https://github.com/deemru/Chromium-Gost/issues/36#issuecomment-886915040 ?

Просто в текущей версии (114.0.5735.110) хотя бы сохраняется залогиненная сессия между перезапусками. А до этого из коробки не работала и синхронизация и сессия не сохранялась.

Если выбирать между двух зол, то лучше пусть из коробки полноценно работают веб-сервисы Google.

Вы же, как продвинутый пользователь, всё так же можете указать недостающие переменные при страте: https://www.chromium.org/developers/how-tos/api-keys/#providing-keys-at-runtime

**Comment by hooddy** [comment]
Да, на этой версии в Гугол просто не зайти в принципе. Но если про меньшее из зол, то ок.

**Comment by deemru** [comment]
> Да, на этой версии в Гугол просто не зайти в принципе.

Зайти, ссылку "как" привёл выше, повторяю: https://www.chromium.org/developers/how-tos/api-keys/#providing-keys-at-runtime

**Comment by UnclWish** [comment]
> Видимо вы продвинутый пользователь, расскажите, как вам удавалось раньше подключать синхронизацию?
> 
> Через вот этот трюк [#36 (comment)](https://github.com/deemru/Chromium-Gost/issues/36#issuecomment-886915040) ?
> 
> Просто в текущей версии (114.0.5735.110) хотя бы сохраняется залогиненная сессия между перезапусками. А до этого из коробки не работала и синхронизация и сессия не сохранялась.
> 
> Если выбирать между двух зол, то лучше пусть из коробки полноценно работают веб-сервисы Google.
> 
> Вы же, как продвинутый пользователь, всё так же можете указать недостающие переменные при старте: https://www.chromium.org/developers/how-tos/api-keys/#providing-keys-at-runtime

Да, я давно присоединился к группе тестировщиков синхронизации. И у меня на всех хромиумных браузерах синхронизация работает без проблем.
И ничего у меня не слетало. Уже год как обновляю Chromium Gost и синхронизация всегда остаётся. Как после перезапуска, так и после обновления.
Вот 110-я версия 1-я когда у меня тупо нет возможности войти в акк даже.

А по поводу недостающих переменных при старте - я, честно говоря, не настолько продвинутый пользователь. По ссылке почитал, ничего не понял. Где прописывать эти гугло-переменные и где брать их значения?

**Comment by deemru** [comment]
> Да, я давно присоединился к группе тестировщиков синхронизации.

Ясно. Вы в числе счастливчиков.

> А по поводу недостающих переменных при старте - я, честно говоря, не настолько продвинутый пользователь. По ссылке почитал, ничего не понял. Где прописывать эти гугло-переменные и где брать их значения?

А теперь представьте, что у вас не работает и синхронизация и ещё при каждом запуске браузера надо вводить логин/пароль при пользовании Gmail или Youtube.

Именно с этой проблемой и была попытка побороться. Сейчас браузера способен работать с сервисами Google как обычный браузер для всех пользователей.

А счастливчикам придётся разобраться как сгенерировать себе GOOGLE_DEFAULT_CLIENT_ID и GOOGLE_DEFAULT_CLIENT_SECRET, если они хотят продолжать пользоваться синхронизацией. Дока есть. Далее устанавливаем переменные и запускаем браузер.

Кстати, возможно правильность этих значений не особо важна, например можно попробовать на Windows:
```cmd
set GOOGLE_DEFAULT_CLIENT_ID=test
set GOOGLE_DEFAULT_CLIENT_SECRET=test
chrome.exe
```

**Comment by UnclWish** [comment]
Со значениями test синхронизация появляется, но показывает, что она приостановлена, надо войти снова. Пытаюсь войти снова, ввожу логин-пароль, но синхронизация не возобновляется.
Видимо, надо туда, всё-таки подставлять правильные значения...

**Comment by UnclWish** [comment]
Проделал все процедуры из статьи: https://www.chromium.org/developers/how-tos/api-keys/#providing-keys-at-runtime
Добавил переменные среды через батник:

setx GOOGLE_API_KEY mykey
setx GOOGLE_DEFAULT_CLIENT_ID myid
setx GOOGLE_DEFAULT_CLIENT_SECRET mysecret

значения подставил свои, полученные по инструкции в статье.

Синхронизация 1-й запуск 110-й версии стала приостановлена, повторил вход - заработала.

Но, блин, это заняло полчаса времени целых. И статься на английском... Хорошо, я немного его знаю.

**Comment by deemru** [comment]
> Но, блин, это заняло полчаса времени целых. И статься на английском... Хорошо, я немного его знаю.

ЧТД -- вы продвинутый пользователь, буду вас упоминать в качестве примера "у кого получилось".

Считаю вопрос решён?

**Comment by UnclWish** [comment]
Да, вопрос решен на данный момент. Спасибо за помощь!

**Comment by UnclWish** [comment]
Сегодня почему-то слетела авторизация в аккаунт Гугла в браузере. Написано - Приостановлена. Повторный вход не помогает. В самом гугле вход есть.
API все прописаны, как и были. Как починить?

**Comment by deemru** [comment]
> Сегодня почему-то слетела авторизация в аккаунт Гугла

Что в `chrome://signin-internals/`?

**Comment by UnclWish** [comment]
Тут много всего:
`Basic Information
Account Consistency	DICE	 
Signin Status	Signed In, Consented for Sync	 
TokenService Load Status	Load credentials finished with success	 
Gaia cookies state	Allowed	 
Account Id	103465898070046199682	 
Gaia Id	103465898070046199682	 
Username	
Auth Error	Invalid credentials (credentials rejected by server).	 
Auth Error Account Id	103465898070046199682	 
Auth Error Username	
Account Reconcilor blocked	True	 
Account Reconcilor State	Scheduled	 
Network calls delayed	False	 
Last Sign-out Source		 
 
Last Signin Details
Gaia Authentication Result	02.08.2018, 19:18:07	Successful
RefreshToken Received	Missing authorization code due to OAuth outage in Dice.	2026-09-09T08:20:38.712Z
Sign-in Access Point		 
 
Access Token Details By Account
103465898070046199682
Service	Requested Scopes	Request Time	Request Status
cloud_policy_client_registration	https://www.googleapis.com/auth/chromeosdevicemanagement
https://www.googleapis.com/auth/userinfo.email
2026-09-09T08:19:09.532Z	
                Failure: Invalid credentials (credentials rejected by server).
              
trusted_vault_frontend	https://www.googleapis.com/auth/cryptauth
2026-09-09T08:19:09.532Z	
                Failure: Invalid credentials (credentials rejected by server).
              
sync	https://www.googleapis.com/auth/chromesync
2026-09-09T08:19:09.532Z	
                Waiting for response
              
oauth_multilogin_token_request	https://www.google.com/accounts/OAuthLogin
2026-09-09T08:19:18.461Z	
                Failure: Invalid credentials (credentials rejected by server).
              
account_capabilities_fetcher	https://www.googleapis.com/auth/account.capabilities
2026-09-09T08:21:22.229Z	
                Failure: Invalid credentials (credentials rejected by server).
              
gaia_account_tracker	https://www.googleapis.com/auth/userinfo.email
https://www.googleapis.com/auth/userinfo.profile
2026-09-09T08:21:22.229Z	
                Failure: Invalid credentials (credentials rejected by server).
              
 
Accounts in Cookie Jar
Email Address	Gaia ID	Validity	Sign-in status	Verification status
	103465898070046199682	Valid	Signed in	Verified
 
Accounts in Token Service
Account Id	Has refresh token	Has persistent auth error	Is bound to the device
103465898070046199682	true	true	false
 
Refresh token events
Timestamp	Accound Id	Event type	Source
2026-09-09T08:19:09.531Z	103465898070046199682	Updated	TokenService::LoadCredentials
2026-09-09T08:19:09.532Z		Loaded	
 
Bound sessions
Session ID	Domain	Path	Expiration time	Throttling Paused	Bound Cookie Names	Refresh URL
gapsts_session	accounts.google.com	/	
                Expires at 2027-10-14T08:20:25.185Z
              	false	__Host-GAPSTS	https://accounts.google.com/RotateBoundGaps
sidts_session	google.com	/	
                Expires at 2026-09-09T08:30:24.938Z
              	false	__Secure-1PSIDRTS, __Secure-1PSIDTS, __Secure-3PSIDRTS, __Secure-3PSIDTS	https://accounts.google.com/RotateBoundCookies`

**Comment by deemru** [comment]
> Failure: Invalid credentials (credentials rejected by server)

С этим на стороне браузера тяжело бороться.

**Comment by UnclWish** [comment]
Утром же всё работало ))) Что случилось, интересно...
Совсем способа нет?

**Comment by UnclWish** [comment]
У меня на домашнем компе стоит SRWare Iron на базе Хрома 150. Там всё работает, только что проверил. И там chrome://signin-internals/ выдаёт другое - там всё хорошо.
Нельзя как-то оттуда что-то перенести?

**Comment by UnclWish** [comment]
Уже новые OAuth client ID прописал и в переменные поставил. Не помогает...

**Comment by UnclWish** [comment]
Попробовал удалить WebData из папки профиля, чтобы сбросить токены. Теперь ошибка Invalid credentials (credentials missing)
Что за ерунда? Как победить?

**Comment by UnclWish** [comment]
Блин, что я только не делал. И профиль удалял, и перестанавливал браузер. Не обновляется токен авторизации и всё тут.
А если новый профиль, то токена вообще нет просто и всё.
Подскажите, люди добрые, как вернуть синхронизацию? Может КриптоПро как-то влияет?

Или это связано с тем, что я в настройках аккаунта гугла вчера включил двухфакторную верификацию и создал ключ доступа для этого компа в Windows Hello... Теперь для подтверждения входа ввожу пароль Hello и в аккаунт гугла на портале входит без проблем. А вот браузер для входа в синхронизацию доступ не получает... Да еще и сбрасывает вход в гугл каждый запуск браузера.
